src/Microsoft.AspNet.Mvc.Core/ControllerActionInvoker.cs

```diff
@@ -83,6 +85,14 @@ protected override void ReleaseInstance(object instance)
             return _argumentBinder.GetActionArgumentsAsync(context, bindingContext);
         }
 
+        protected override Task ValidateActionArgumentsAsync(
+            ActionContext context, 
+            ActionBindingContext bindingContext, 
+            IDictionary<string, object> actionArguments)
+        {
+            return _controllerActionArgumentValidator.ValidateArgumentsAsync(context, bindingContext, actionArguments);
+        }
```

Comment:
Well, the argument binder looks for things like binder metadata ( the 4 interfaces) etc, going forward that code will also contain stuff for controller properties etc. It seemed more cleaner to have a clear separation of two stages ( was also thinking about a visitor pattern like the one you suggested earlier and it would be easier to do it in future with the current design). Let me know if you disagree.